### ISSUE TITLE
Debug Adapter Protocol support

### ISSUE THREAD

**Issue by fwcd** [Microsoft/language-server-protocol-inspector#38]
Would it be possible to support the [Debug Adapter Protocol](https://microsoft.github.io/debug-adapter-protocol/) too?

**Comment by octref** [comment]
I don't really know much about DAP. I guess the UI can be reused. What would you use it for?

**Comment by fwcd** [comment]
I am building [a debug adapter for Kotlin](https://github.com/fwcd/KotlinDebugAdapter) where being able to visualize the flow of JSON messages (for debugging purposes) would be really useful.

**Comment by weinand** [comment]
@fwcd yes, the language-server-protocol-inspector wold be useful for DAP too (since DAP is conceptually equivalent to LSP).

A nice feature of VS Code is that it provides an extension API for accessing any DAP flow at runtime. E.g. this code outputs the DAP flow for for all "node" debug sessions:
```js
vscode.debug.registerDebugAdapterTracker('node', session => {
    return {
        fromDebugAdapter: m => console.log(`<=== ${JSON.stringify(m, undefined, 2)}`),
        toDebugAdapter: m => console.log(`===> ${JSON.stringify(m, undefined, 2)}`),
    };
});
```

So having an extension that provides a UI on top of the code from above would be nice.
